test/future-test.js

```diff
@@ -50,196 +53,462 @@ describe('future', () => {
-
 				const p = never()
 				resolve(p)
 				return assert.strictEqual(p, promise.ap(fulfill(1)))
 			})
 		})
 
 		describe('concat', () => {
-			it('should behave like fulfilled', () => {
+			it('should be identity for fulfill', () => {
 				const { resolve, promise } = future()
-
 				const p1 = fulfill(1)
 				const p2 = fulfill(2)
-
 				resolve(p1)
-				return assertSame(p1.concat(p2), promise.concat(p2))
+				assert.strictEqual(p1, promise.concat(p2))
 			})
 
-			it('should behave like rejected', () => {
+			it('should be identity for reject', () => {
 				const { resolve, promise } = future()
+				const p1 = silenced(reject(new Error()))
+				const p2 = silenced(reject(new Error()))
+				resolve(p1)
+				assert.strictEqual(p1, promise.concat(p2))
+			})
 
-				const p1 = reject(new Error())
-				const p2 = reject(new Error())
-				silenceError(p1)
-				silenceError(p2)
-
+			it('should return other for never', () => {
+				const { resolve, promise } = future()
+				const p1 = never()
+				const p2 = fulfill(2)
 				resolve(p1)
 				assert.strictEqual(p1.concat(p2), promise.concat(p2))
 			})
+		})
+	})
+
+	describe('before being resolved to another promise', () => {
+		describe('state', () => {
+			it('should be pending', () => {
+				const { promise } = future()
+				assert(isPending(promise))
+			})
 
-			it('should behave like never', () => {
+			it('should not be settled', () => {
+				const { promise } = future()
+				assert(!isSettled(promise))
+			})
+		})
+
+		describe('inspect', () => {
+			it('should not be fulfilled', () => {
+				const { promise } = future()
+				assert.notStrictEqual(fulfill().inspect(), promise.inspect())
+			})
+
+			it('should not be rejected', () => {
+				const { promise } = future()
+				assert.notStrictEqual(silenced(reject()).inspect(), promise.inspect())
+			})
+		})
+
+		describe('then', () => {
+			it('should behave like mapped for fulfill', () => {
 				const { resolve, promise } = future()
+				const p = fulfill(1)
+				const res = promise.then(f)
+				resolve(p)
+				return assertSame(p.map(f), res)
+			})
 
-				const p1 = never()
-				const p2 = fulfill(2)
+			it('should behave like chained for fulfill', () => {
+				const { resolve, promise } = future()
+				const p = fulfill(1)
+				const res = promise.then(fp)
+				resolve(p)
+				return assertSame(p.chain(fp), res)
+			})
 
-				resolve(p1)
-				return assertSame(p1.concat(p2), promise.concat(p2))
+			it('should behave like rejection chained for fulfill', () => {
+				const { resolve, promise } = future()
+				const p = fulfill(1)
+				const res = promise.then(rp)
+				resolve(p)
+				return assertSame(p.chain(rp), res)
+			})
+
+			it('should behave like rejected for reject', () => {
+				const { resolve, promise } = future()
+				const p = silenced(reject(1))
+				const res = promise.then(f)
+				resolve(p)
+				return assertSame(p, res)
+			})
+
+			/* it('should have never state for never', () => {
```

Comment:
They're commented out because they all fail :-) See #30
This is a good example for a difference between futures that are resolved before the method call and those that are resolved "asynchronously".